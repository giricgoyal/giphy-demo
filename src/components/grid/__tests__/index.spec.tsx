import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import Grid, { Props } from '../index'
import { setupIntersectionObserverMock } from '../../../mocks'

describe('<Grid />', () => {
    let container

    const defaultProps: Props = {
        isLoading: true,
        columns: 3,
        headerIcon: 'cross',
    }

    beforeEach(() => {
        setupIntersectionObserverMock()
    })

    describe('When isLoading is true', () => {
        beforeEach(() => {
            ;({ container } = render(<Grid {...defaultProps}></Grid>))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When isLoading is false', () => {
        beforeEach(() => {
            const props: Props = {
                ...defaultProps,
                isLoading: false,
            }

            ;({ container } = render(
                <Grid {...props}>
                    <div>No time to die</div>
                    <div>Spectre</div>
                    <div>Casino Royale</div>
                    <div>The man with the golden gun</div>
                    <div>Dr. No</div>
                    <div>For your eyes only</div>
                    <div>License to kill</div>
                </Grid>,
            ))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })
})
