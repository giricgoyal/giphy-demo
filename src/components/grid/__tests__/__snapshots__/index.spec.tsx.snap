// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> When isLoading is false should match snapshot 1`] = `
<div>
  <div
    class="grid"
    data-testid="grid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div>
          No time to die
        </div>
        <div>
          The man with the golden gun
        </div>
        <div>
          License to kill
        </div>
      </div>
      <div
        class="col"
      >
        <div>
          Spectre
        </div>
        <div>
          Dr. No
        </div>
      </div>
      <div
        class="col"
      >
        <div>
          Casino Royale
        </div>
        <div>
          For your eyes only
        </div>
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`<Grid /> When isLoading is true should match snapshot 1`] = `
<div>
  <div
    class="grid"
    data-testid="grid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      />
      <div
        class="col"
      />
      <div
        class="col"
      />
    </div>
    <div
      class="grid__loader"
    >
      <div
        class="spinner-border"
      />
    </div>
  </div>
</div>
`;
